### PR TITLE
Feature lwdev 9718 multiple deposits

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/BlockchainCashoutPreconditionsCheckClient.cs [hunk 1]
```diff
@@ -33,28 +34,21 @@ public BlockchainCashoutPreconditionsCheckClient(string serviceUrl)
...
-                case CashoutValidityResult validationErrorResponse:
-                    isAllowed = validationErrorResponse.IsAllowed;
-                    validationErrors = validationErrorResponse.ValidationErrors;
-                    break;
-                case ErrorResponse errorResponse:
-                    validationErrors = new[] { new ValidationErrorResponse(ValidationErrorType.None, errorResponse.ErrorMessage) };
-                    break;
-                default:
-                    throw new Exception($"Unknown response: {responseObject.ToJson()}");
+                validationErrors = new[] { ValidationErrorResponse.Create(ValidationErrorType.None, ""/*errorResponse.ErrorMessage*/), };
```
Comment (KonstantinRyazantsev) [inline client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/BlockchainCashoutPreconditionsCheckClient.cs]: why ErrorMessages is commented out?

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/ClientGenerator/BlockchainSettingsClientFactory.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+namespace Lykke.Service.BlockchainWallets.Client.ClientGenerator
+{
+    public class BlockchainCashoutPreconditionsApiFactory : IBlockchainCashoutPreconditionsApiFactory
+    {
+        public IBlockchainCashoutPreconditionsCheckApi CreateNew(BlockchainCashoutPreconditionsClientSettings settings,
+            bool withCaching = true,
+            IClientCacheManager clientCacheManager = null,
+            params DelegatingHandler[] handlers)
+        {
+            return CreateNew(settings?.ServiceUrl, withCaching, clientCacheManager, handlers);
```
Comment (KonstantinRyazantsev) [inline client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/ClientGenerator/BlockchainSettingsClientFactory.cs]: Why settings**?**.ServiceUrl ? Probably better to check that settings is not null and throw an exception if it is.

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.DelegatingHandlers
+{
+    //Used in tests only to redirect http requests to the test fixture server
+    public class RequestInterceptorHandler : DelegatingHandler
```
Comment (KonstantinRyazantsev) [inline client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs]: If it's used only in tests, better to declare it in tests assembly

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            CancellationToken cancellationToken)
+        {
+            var clonedRequest = await request.CloneAsync();
+            var response = await _httpClient.SendAsync(clonedRequest, cancellationToken);
+
+            return response;
+        }
+    }
+
+    internal static class HttpRequestMessageTestExtension
```
Comment (KonstantinRyazantsev) [inline client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs]: Multiple classes in the same file

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.csproj [hunk 5]
```diff
@@ -24,13 +24,15 @@
...
+    <PackageReference Include="Lykke.Common" Version="7.1.0" />
+    <PackageReference Include="Lykke.HttpClientGenerator" Version="2.3.0" />
     <PackageReference Include="MessagePack" Version="1.7.3.4" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.11" />
+    <PackageReference Include="Refit" Version="4.6.30" />
   </ItemGroup>
 
   <ItemGroup>
-    <Folder Include="AutorestClient\" />
+    <ProjectReference Include="..\..\contract\Lykke.Service.BlockchainCashoutPreconditionsCheck.Contract\Lykke.Service.BlockchainCashoutPreconditionsCheck.Contract.csproj" />
```
Comment (KonstantinRyazantsev) [inline client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.csproj:35]: Is contract published via separate nuget? If not, then extra steps are required to include both assemblies to the Client nuget

File: tests/Lykke.Service.BlockchainCashoutPreconditionsCheck.Test/Program.cs [hunk 6]
```diff
@@ -1,43 +1,42 @@
...
         {
-            Task.Delay(10000).Wait();
-            var client = new Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.BlockchainCashoutPreconditionsCheckClient("http://localhost:5000", null);
-            var (isAllowed, listOfErrors) = client.ValidateCashoutAsync(new Client.Models.CashoutValidateModel()
-            {
-                AssetId = "62c04960-4015-4945-bb7e-8e4a193b3653",
-                DestinationAddress = "0x406561f72e25ab41200fa3d52badc5a21",
-                Amount = 0
-            }).Result;
+            //Task.Delay(10000).Wait();
```
Comment (KonstantinRyazantsev) [inline tests/Lykke.Service.BlockchainCashoutPreconditionsCheck.Test/Program.cs]: Let's remove this project at all

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/BlockchainCashoutPreconditionsCheckClient.cs (hunk 1 above)
Comment (OttoVT) [reply client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/BlockchainCashoutPreconditionsCheckClient.cs]: fixed

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/ClientGenerator/BlockchainSettingsClientFactory.cs (hunk 2 above)
Comment (OttoVT) [reply client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/ClientGenerator/BlockchainSettingsClientFactory.cs]: fixed

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs (hunk 3 above)
Comment (OttoVT) [reply client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs]: fixed

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs (hunk 4 above)
Comment (OttoVT) [reply client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/DelegatingHandlers/RequestInterceptorHandler.cs]: fixed

File: client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.csproj (hunk 5 above)
Comment (OttoVT) [reply client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client/Lykke.Service.BlockchainCashoutPreconditionsCheck.Client.csproj:35]: image

File: tests/Lykke.Service.BlockchainCashoutPreconditionsCheck.Test/Program.cs (hunk 6 above)
Comment (OttoVT) [reply tests/Lykke.Service.BlockchainCashoutPreconditionsCheck.Test/Program.cs]: Fixed

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**